Add customizable notifications for implings

DapperMickie opened the pull request:
As discussed here:
https://github.com/runelite/runelite/issues/17799#issuecomment-2081159199

Adam- commented:
I don't think this makes sense because there is still impling mode NOTIFY despite a separate notification option existing. This means you could enable the notification but still not receive notifications due to not setting the *other setting* to notify. Also the comment you have linked to is just me saying I don't know how this should work.

DapperMickie commented:
Added a checkbox for highlighting the implings.
Added a Notification for implings (instead of the customizableNotification that didn't make much sense).
Added a migration flow for configs.
